scipy/optimize/cobyla.py

```diff
@@ -22,9 +22,23 @@
     izip = zip
 
 
-__all__ = ['fmin_cobyla']
+# Workaround as _cobyla.minimize is not threadsafe
+# due to an unknown f2py bug and can segfault, 
+# see gh-9658.
+
+from threading import Lock
+_module_lock = Lock()
```

Comment:
RLock, f2py is reentrant

Reply:
It should be, but it is hard to say if it is the Fortran or C code that segfaults. Anyway, we are not doing any recursive calls here so both will work. I have changed it to RLock but I do not think it matters.